common/src/reserved.rs

```diff
@@ -338,4 +462,189 @@ mod tests {
                 .collect::<HashSet<_>>()
         );
     }
+
+    fn setup_TxDelegate_test() -> (PublicKey, PrivateKey, PublicKey, Member, ReservedState) {
+        let keys = (0..4)
+            .into_iter()
+            .map(|i| generate_keypair(format!("{}", i)))
+            .collect::<Vec<_>>();
+
+        let delegator_public_key = keys[0].0.clone();
+        let delegator_private_key = keys[0].1.clone();
+        let delegatee_public_key = keys[1].0.clone();
+
+        let delegator = Member {
+            public_key: delegator_public_key.clone(),
+            name: "delegator".to_string(),
+            consensus_voting_power: 10,
+            governance_voting_power: 10,
+            consensus_delegations: None,
+            governance_delegations: None,
+        };
+
+        let delegatee = Member {
+            public_key: delegatee_public_key.clone(),
+            name: "delegatee".to_string(),
+            consensus_voting_power: 20,
+            governance_voting_power: 20,
+            consensus_delegations: None,
+            governance_delegations: None,
+        };
+
+        let members = vec![
+            delegator.clone(),
+            delegatee.clone(),
+            create_member(keys.clone(), 2),
+        ];
+
+        let genesis_header = BlockHeader {
+            author: PublicKey::zero(),
+            prev_block_finalization_proof: Vec::new(),
+            previous_hash: Hash256::zero(),
+            height: 0,
+            timestamp: 0,
+            commit_merkle_root: Hash256::zero(),
+            repository_merkle_root: Hash256::zero(),
+            validator_set: members
+                .iter()
+                .map(|member| (member.public_key.clone(), member.consensus_voting_power))
+                .collect::<Vec<_>>(),
+            version: "0.1.0".to_string(),
+        };
+
+        let genesis_info = GenesisInfo {
+            header: genesis_header.clone(),
+            genesis_proof: keys
+                .iter()
+                .map(|(_, private_key)| TypedSignature::sign(&genesis_header, private_key).unwrap())
+                .collect::<Vec<_>>(),
+            chain_name: "test-chain".to_string(),
+        };
+
+        let mut state = ReservedState {
+            genesis_info,
+            members: vec![delegator.clone(), delegatee.clone()],
+            consensus_leader_order: vec![delegator.name.to_string(), delegatee.name.to_string()],
+            version: "".to_string(),
+        };
+        (
+            delegator_public_key,
+            delegator_private_key,
+            delegatee_public_key,
+            delegatee,
+            state,
+        )
+    }
+
+    #[test]
+    fn test_apply_delegate_with_governance_success() {
+        // given
+        setup_test();
+        let (
+            delegator_public_key,
+            delegator_private_key,
+            delegatee_public_key,
+            delegatee,
+            mut state,
+        ) = setup_TxDelegate_test();
```

Comment:
The method is designated for setting up the `given` environment for `TxDelegate` unit tests.